source/MicroBitMemoryMap.cpp

```diff
@@ -138,5 +140,58 @@ void MicroBitMemoryMap::findHashes()
                 memoryMapStore.memoryMap[2].startAddress = (uint32_t)magicAddress;
                 return;
         }
+        
+        if (   *(magicAddress - (sizeof(magicAddress))) == 0x597F30FE
+            && *(magicAddress - (sizeof(magicAddress)/4))   == 0xC1B1D79D
+           ) {
+            // Found uPy Magic
+            DMESG("FOUND UPY_MAGIC");
+            uint8_t nRegions = (*(magicAddress - (sizeof(magicAddress)/2)));
+            
+            // Iterate through regions
+            for(int i = 0; i < nRegions; i++) {
+               processRecord(magicAddress - ((2 + i) * sizeof(magicAddress))); 
+            }
+
+        }
     }
 }
+
+
+/*
+ * Function to process record from uPy build
+ */
+void MicroBitMemoryMap::processRecord(uint32_t *address) {
+    uint8_t id = (*address) - 1;
+    uint8_t hashType = ((*address) >> 8) & 0xFF;
+
+    uint32_t start; memcpy(&start, address, 4); 
+    start = (start & 0xFFFF0000) >> 8;
+    start = start * 4096;
+
+    uint32_t length; memcpy(&length, address + 1, 4); 
+
+    uint32_t hash[2]; 
+    if(hashType == 2) {
+        memcpy(&hash, (uint32_t *) *(address + 2), 8);
+    } else {
+        memcpy(&hash, address + 2, 8);
+    }
```

Comment:
It'd be good to add a note about what the different hash types are, unless it's already covered somewhere else.
An enum with the note would be ideal.

Reply:
👍